ibis/backends/pyspark/compiler.py

```diff
@@ -1532,8 +1543,15 @@ def _get_interval_col(
             'Interval unit "{}" is not allowed. Allowed units are: '
             '{}'.format(dtype.unit, allowed_units)
         )
+
+    if isinstance(op.value, pd.Timedelta):
+        pandas_unit = _time_unit_mapping_pd_timedelta[dtype.unit]
```

Comment:
I am not sure we need this. Does pyspark not work with pd.Timedelta directly?

Reply:
If we just pass in op.value as the existing code does, then Pyspark blows up with a ParseException, e.g. if we run 

```
t.timestamp_col + pd.Timedelta('10s')
```
the error is

```
E   pyspark.sql.utils.ParseException: 
E   mismatched input ':' expecting {'ADD', 'AFTER', 'ALL', 'ALTER'...
<truncated>
E   
E   == SQL ==
E   INTERVAL 0 days 00:00:10 second
E   ------------------^^^

```
so at least if we want to use this `INTERVAL` syntax, Pyspark doesn't do well with the colons in the Pandas Timedelta representation.

Reply:
kk